clients/v1_auth/v1_auth.go

```diff
@@ -0,0 +1,423 @@
+package v1_auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+)
+
+type Client struct {
+	clients.CLI
+	api.LegacyAuthorizationsApi
+	api.UsersApi
+	api.OrganizationsApi
+}
+
+type v1PrintOpts struct {
+	deleted bool
+	token   *v1Token
+	tokens  []v1Token
+}
+
+type v1Token struct {
+	ID          string   `json:"id"`
+	Description string   `json:"description"`
+	Token       string   `json:"token"`
+	Status      string   `json:"status"`
+	UserName    string   `json:"userName"`
+	UserID      string   `json:"userID"`
+	Permissions []string `json:"permissions"`
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	Desc        string
+	Username    string
+	Password    string
+	NoPassword  bool
+	ReadBucket  []string
+	WriteBucket []string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if params.Password != "" && params.NoPassword {
+		return fmt.Errorf("only one of --password and --no-password may be specified")
+	}
+
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	// verify an existing token with the same username doesn't already exist
+	auths, err := c.LegacyAuthorizationsApi.GetAuthorizations(ctx).Token(params.Username).Execute()
+	if apiError, ok := err.(api.ApiError); ok {
+		if apiError.ErrorCode() != api.ERRORCODE_NOT_FOUND {
+			return fmt.Errorf("failed to verify username %q has no auth: %w", params.Username, err)
+		} else if len(auths.GetAuthorizations()) != 0 {
+			return fmt.Errorf("authorization with username %q already exists", params.Username)
+		}
+	}
+	auth := auths.GetAuthorizations()[0]
```

Comment:
This line will always fail, since we're asserting above that `len(auths.GetAuthorizations()) == 0`